pywbem_mock/_wbemconnection_mock.py

```diff
@@ -995,6 +1009,32 @@ def _display_objects(obj_type, object_repo, namespace, cmt_begin, cmt_end,
                     else:
                         _uprint(dest, obj.tomof())
 
+    def register_provider(self, namespace, classname, provider):
+        """
+        Register the provider object for namespace and class.
+        TODO: The registration data must be in the providers somewhere since
+        there is no link back to this class from the providers.
+
+        Parameters:
+
+          namespace ():
+
+          classname ():
+
+          provider ():
+        """
+        self.mainprovider.validate_namespace(namespace)
+
+        assert classname is not None
+
+        # TODO validate that classname is in repository
+
+        if namespace not in self.provider_registry:
+            self.provider_registry[namespace] = namespace
+        if classname not in self.provider_registry[namespace][classname]:
+            self.provider_registry[namespace][classname] = provider
+        print("PROVIDERREG %r" % self.provider_registry)
```

Comment:
I think what you want are two levels of dicts: key namespace -> key classname -> value provider object.
If so, the logic in the lines above seems wrong. Did you mean this:
```
       if namespace not in self.provider_registry:
            self.provider_registry[namespace] = NocaseDict()
        if classname not in self.provider_registry[namespace]:
            self.provider_registry[namespace][classname] = provider
```

Reply:
You are right.   FIX

Reply:
I just concluded late last night the the design of the provider registery should be:

dict of classes
       [namespaces] : provider

and the register_provider parameters are list of namespaces and list of classes.

The class must be there. The namespaces is optional

Further the provider  can be expanded to "instance provider", "method "provider" 

WILL fix this morning and upload along with first examples